src/api/cpp/nixl_v2_2.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved. SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _NIXL_H
+#define _NIXL_H
+
+#include <cstdint>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace nixl {
+
+class Config {
+};
+
+class MemDesc {
+public:
+    enum class MemType { DRAM, VRAM, BLK, OBJ, FILE, PODMEM };
+
+    // Local memory
+    MemDesc(uint64_t address, size_t length, MemType type, uint64_t dev_id);
+
+    // Remote memory
+    MemDesc(uint64_t address, size_t length, MemType type, uint64_t dev_id,
+            std::string_view agent_name);
+};
+
+class Agent {
```

Comment:
I recommend using an abstract class as an interface. E.g.,
```C++
class IAgent {
public:
    using Ptr = std::unique_ptr<IAgent>;

    virtual ~IAgent() = default;
    virtual XferRequest postXfer(const XferList &xferList) = 0;
    virtual XferRequest postXfer(const XferList &xfer_list, const XferIndices &xferIndices) = 0;
    // etc.
};
```

And separate creation of the object from the interface.
```C++
IAgent::Ptr createAgent(const Config &config);
```

Reply:
agree, when I wrote the comment, I used `std::unique_ptr<IAgent>` several times, so decided to add an alias. But now it is redundant.